app/src/kscan_sideband_behaviors.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_kscan_sideband_behaviors
+
+#include <zephyr/device.h>
+#include <drivers/behavior.h>
+#include <zephyr/drivers/kscan.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+
+#include <zmk/event_manager.h>
+#include <zmk/behavior.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct ksbb_entry {
+    uint8_t row;
+    uint8_t column;
+    struct zmk_behavior_binding binding;
+};
+
+struct ksbb_config {
+    const struct device *kscan;
+    struct ksbb_entry *entries;
+    size_t entries_len;
+};
+
+struct ksbb_data {
+    kscan_callback_t callback;
+    bool enabled;
+};
+
+#define GET_KSBB_DEV(n) DEVICE_DT_GET(DT_DRV_INST(n)),
+
+// The kscan callback has no context with it, so we keep a static array of all possible
+// KSBBs to check when a kscan callback from the "wrapped" inner kscan fires.
+static const struct device *ksbbs[DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)] = {
```

Comment:
Specifying the array size shouldn't be necessary?
```suggestion
static const struct device *ksbbs[] = {
```